components/molecule/modal/src/HoC/WithUrlState.js

```diff
@@ -0,0 +1,69 @@
+import React, {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+
+const getUrl = ({hash: hashName} = {}) => {
+  const location = window.location
+  const hash = hashName ? `#${hashName}` : ''
+  return `${location.origin}${location.pathname}${location.search}${hash}`
+}
+
+const checkHash = hash => window.location.hash.includes(hash)
+
+const emptyFn = () => {}
+
+/**
+ * It must be refactored with react-router hooks (in React Router > 5)
+ * - useHistory
+ * - useLocation
+ */
+export default BaseComponent => {
+  const displayName = BaseComponent.displayName
+
+  if (typeof window === 'undefined') return BaseComponent
+
+  const WithUrlState = props => {
+    const {hash, openModalTrigger, onClose, ...rest} = props
+    const {isOpen} = rest
```

Comment:
why don't you get it from props `destructuring` as in L25?

Reply:
I destructure `rest` in BaseComponent so it should keep there.